### PR TITLE
Widen attention support for HRX for llama 3.3 8b

### PR DESCRIPTION
**Pull request by rsuderman**
Extend the HRX F32/F16 FLASH_ATTN_EXT decode path to accept KV lengths up to 8192 by using a streaming/recompute path when logits no longer fit in the existing 1024-entry shared buffer.

Keep other decode K/V types capped at 1024, add focused HRX support tests for the current and future long-KV matrix, and add backend-ops coverage for grouped-query D=128 flash attention shapes.

### CODE REVIEW

File: tests/test-backend-hrx.cpp [hunk 1]
```diff
@@ -3520,7 +3424,272 @@ static void run_flash_attn_ext_prefill_f16_case(
...
+    std::vector<float> v_data(static_cast<size_t>(D * KV * H_KV * S));
+    for (size_t i = 0; i < q_data.size(); ++i) {
+        q_data[i] = static_cast<float>(static_cast<int>((i * 7 + 3) % 41) - 20) / 61.0f;
+    }
+    for (size_t i = 0; i < k_data.size(); ++i) {
+        k_data[i] = static_cast<float>(static_cast<int>((i * 11 + 5) % 37) - 18) / 67.0f;
+    }
+    for (size_t i = 0; i < v_data.size(); ++i) {
+        v_data[i] = static_cast<float>(static_cast<int>((i * 13 + 9) % 43) - 21) / 59.0f;
+    }
```
Comment (AaronStGeorge) [inline tests/test-backend-hrx.cpp:3721]: A comment explaining why these map to reasonable activations would be nice.

**Review by AaronStGeorge — APPROVED** [review]

File: tests/test-backend-hrx.cpp (hunk 1 above)
Comment (rsuderman) [reply tests/test-backend-hrx.cpp:3721]: Comments added.
